src/main/scala/io/delta/DeltaMergeBuilder.scala

```diff
@@ -132,31 +379,107 @@ case class DeltaMergeMatchedActionBuilder private[delta](
   }
 }
 
-case class DeltaMergeNotMatchedActionBuilder private[delta](
+object DeltaMergeMatchedActionBuilder {
+  /**
+   * :: Unstable ::
+   *
+   * Private method for internal usage only. Do not call this directly.
+   */
+  @Unstable
+  private[delta] def apply(
+      mergeBuilder: DeltaMergeBuilder,
+      matchCondition: Option[Column]): DeltaMergeMatchedActionBuilder = {
+    new DeltaMergeMatchedActionBuilder(mergeBuilder, matchCondition)
+  }
+}
+
+
+/**
+ * :: Evolving ::
+ *
+ * Builder class to specify the actions to perform when a source row has not matched any target
+ * Delta table row based on the merge condition, but has matched any additional condition
+ * if specified.
+ *
+ * See [[DeltaMergeBuilder]] for more information.
+ *
+ * @since 0.3.0
+ */
+@Evolving
+class DeltaMergeNotMatchedActionBuilder private(
     private val mergeBuilder: DeltaMergeBuilder,
-    private val matchCondition: Option[Column]) {
+    private val notMatchCondition: Option[Column]) {
 
+  /**
+   * :: Evolving ::
+   *
+   * Insert a new target Delta table row based on the rules defined by `values`.
+   *
+   * @param values rules to insert a row as a Scala map between target column names and
+   *               corresponding expressions as Column objects.
+   * @since 0.3.0
+   */
+  @Evolving
   def insert(values: Map[String, Column]): DeltaMergeBuilder = {
     addInsertClause(values)
   }
 
+  /**
+   * :: Evolving ::
+   *
+   * Insert a new target Delta table row based on the rules defined by `values`.
```

Comment:
Insert new rows to the target table based on the rules defined by `values`.

apply to all overloaders below.